opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -70,6 +70,47 @@
 ParentSpan = typing.Optional[typing.Union["Span", "SpanContext"]]
 
 
+class Link:
+    """A link to a `Span`."""
+
+    def __init__(
+        self, context: "SpanContext", attributes: types.Attributes = None
+    ) -> None:
+        self._context = context
+        self._attributes = attributes
+
+    @property
+    def context(self) -> "SpanContext":
+        return self._context
+
+    @property
+    def attributes(self) -> types.Attributes:
+        return self._attributes
+
+
+class Event:
+    """A text annotation with a set of attributes."""
+
+    def __init__(
+        self, name: str, timestamp: int, attributes: types.Attributes = None
+    ) -> None:
+        self._name = name
+        self._attributes = attributes
+        self._timestamp = timestamp
+
+    @property
```

Comment:
Making these properties in the API means subclasses can't make them regular attributes. Is that intentional?

Reply:
Well, I could say it is, subclasses would have to re-implement this property if needed.

Reply:
Bit they could never choose to make these regular attributes. That is, you can always replace attributes with properties in a subclass, but not vice-versa.